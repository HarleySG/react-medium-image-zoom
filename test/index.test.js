import React, { createRef } from 'react'
import { renderIntoDocument } from 'react-dom/test-utils'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Container, { ImageZoom } from '../source'

describe('ImageZoom', () => {

  describe('container', () => {

    it('renders', () => {
      const wrapper = shallow(
        <Container
        />
      )
      const tree = toJson(wrapper)
      expect(tree).toMatchSnapshot()
    })

    it('handles ref', () => {
      const ref = createRef()
      renderIntoDocument(
        <Container
          ref={ref}
        />
      )
      expect(ref.current).toBeInstanceOf(HTMLDivElement)
    })

  })

  describe('component', () => {

    it('renders, no children', () => {
      const wrapper = shallow(
        <ImageZoom
        />
      )
      const tree = toJson(wrapper)
      expect(tree).toMatchSnapshot()
    })

  })

})
