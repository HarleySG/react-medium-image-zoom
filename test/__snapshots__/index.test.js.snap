// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageZoom component renders, no children 1`] = `
<StrictMode>
  <div>
    Hello, world!
  </div>
</StrictMode>
`;

exports[`ImageZoom container renders 1`] = `
<ImageZoom
  forwardedRef={null}
/>
`;
